v2/internal/reconcilers/arm/arm_client_cache.go

```diff
@@ -242,3 +259,38 @@ func (c *ARMClientCache) getSecret(ctx context.Context, namespace string, secret
 func matchSecretData(old, new map[string][]byte) bool {
 	return reflect.DeepEqual(old, new)
 }
+
+// TODO: Not sure if its a right thing to put this here or create a separate package for identity/auth and put workloadIdentityCredential there?
+// The below bit is from https://github.com/Azure/azure-sdk-for-go/issues/15615#issuecomment-1211012677
```

Comment:
I'd lean towards putting it into a separate package as you suggested.

Reply:
Yep, I'll keep this one as it is for now. And since I'm gonna work more on workload identity stuff which Matt and I had a conversation about(Matt has mentioned above). Basically, the global credential code in `main.go` would be using this bit as well so will move it in next PR.